tests/yakus/dragonpon.test.cpp

```diff
@@ -0,0 +1,105 @@
+#include <gtest/gtest.h>
+#include "hands.h"
+#include "hand.h"
+#include "gamestate.h"
+#include "handformer.h"
+#include "handnode.h"
+#include "analysis.h"
+
+using Mahjong::GameState;
+using Mahjong::Hand;
+using Mahjong::HandFromNotation;
+using Mahjong::breakdownHand;
+using Mahjong::Node;
+using Mahjong::Piece;
+
+TEST(isWindOrDragonPon, WhiteDragon) {
+  auto gameState = GameState();
+  gameState.hands[0] = Hand(HandFromNotation("123m456m555z11z"));
+
+  auto root = breakdownHand(gameState.hands.at(0).live);
+
+  for (const auto& branch : Node::AsBranchVectors(root.get())) {
+    if (Mahjong::isWindOrDragonPon(gameState, 0, branch) == 1) {
+      SUCCEED();
+      return;
+    }
+  }
+  FAIL();
+}
+
+TEST(isWindOrDragonPon, GreenDragon) {
+  auto gameState = GameState();
+  gameState.hands[0] = Hand(HandFromNotation("123m456m666z11z"));
+
+  auto root = breakdownHand(gameState.hands.at(0).live);
+
+  for (const auto& branch : Node::AsBranchVectors(root.get())) {
+    if (Mahjong::isWindOrDragonPon(gameState, 0, branch) == 1) {
+      SUCCEED();
+      return;
+    }
+  }
+  FAIL();
+}
+
+TEST(isWindOrDragonPon, RedDragon) {
+  auto gameState = GameState();
+  gameState.hands[0] = Hand(HandFromNotation("123m456m777z11z"));
+
+  auto root = breakdownHand(gameState.hands.at(0).live);
+
+  for (const auto& branch : Node::AsBranchVectors(root.get())) {
+    if (Mahjong::isWindOrDragonPon(gameState, 0, branch) == 1) {
+      SUCCEED();
+      return;
+    }
+  }
+  FAIL();
+}
+
+TEST(isWindOrDragonPon, Kan) {
+  auto gameState = GameState();
+  gameState.hands[0] = Hand(HandFromNotation("123m456m7777z11z"));
+
+  auto root = breakdownHand(gameState.hands.at(0).live);
+
+  for (const auto& branch : Node::AsBranchVectors(root.get())) {
+    if (Mahjong::isWindOrDragonPon(gameState, 0, branch) == 1) {
+      SUCCEED();
+      return;
+    }
+  }
+  FAIL();
+}
+
+TEST(isWindOrDragonPon, CanWhenOpen) {
+  auto gameState = GameState();
+  gameState.hands[0] = Hand(HandFromNotation("123m456m777z11z"));
+  gameState.hands[0].open = true;
+
+  auto root = breakdownHand(gameState.hands.at(0).live);
+
+  for (const auto& branch : Node::AsBranchVectors(root.get())) {
+    if (Mahjong::isWindOrDragonPon(gameState, 0, branch) == 1) {
+      SUCCEED();
+      return;
+    }
+  }
+  FAIL();
+}
+
+TEST(isWindOrDragonPon, BadHand) {
+  auto gameState = GameState();
+  gameState.hands[0] = Hand(HandFromNotation("123m456m222z11z"));
```

Comment:
Doing a bad hand with this set is strange because it could also fail because of the wind pon but you aren't testing the wind pons here. I would have bad hand contain no honors or combine wind and honors testing.

Reply:
I now see how it's getting a bit confusing with the negative tests. I'll make your former change suggestion, I'm still of the opinion they should stay separate due to the quantity of tests.

Reply:
We should discuss it back in gnv

Reply:
Did former suggestion in 8ca3870ae0bf49165c9fc1220ded49941f8f0ccc

As said above, we should discuss it in gnv